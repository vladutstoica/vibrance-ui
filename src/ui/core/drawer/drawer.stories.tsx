import type { Meta, StoryObj } from "@storybook/react";

import React from "react";
import {
  Drawer,
  DrawerClose,
  DrawerContent,
  DrawerDescription,
  DrawerFooter,
  DrawerHeader,
  DrawerTitle,
  DrawerTrigger,
} from "./drawer";

/**
 * A drawer component for React.
 */
const meta = {
  title: "Design System/Core/Drawer",
  component: Drawer,
  tags: ["autodocs"],
  argTypes: {},
  render: (args) => (
    <Drawer {...args}>
      <DrawerTrigger>Open</DrawerTrigger>
      <DrawerContent>
        <DrawerHeader>
          <DrawerTitle>Are you sure absolutely sure?</DrawerTitle>
          <DrawerDescription>This action cannot be undone.</DrawerDescription>
        </DrawerHeader>
        <DrawerFooter>
          {/* biome-ignore lint/a11y/useButtonType: <explanation> */}
          <button className="rounded bg-primary px-4 py-2 text-primary-foreground">
            Submit
          </button>
          <DrawerClose>
            {/* biome-ignore lint/a11y/useButtonType: <explanation> */}
            <button className="hover:underline">Cancel</button>
          </DrawerClose>
        </DrawerFooter>
      </DrawerContent>
    </Drawer>
  ),
  parameters: {
    layout: "centered",
  },
} as Meta<typeof Drawer>;

export default meta;

type Story = StoryObj<typeof meta>;

/**
 * The default form of the drawer.
 */
export const Default: Story = {} as Story;
