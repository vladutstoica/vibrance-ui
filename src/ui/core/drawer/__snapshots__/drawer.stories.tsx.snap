// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design System/Core/Drawer Default smoke-test 1`] = `
<button type="button"
        aria-haspopup="dialog"
        aria-expanded="false"
        aria-controls="radix-:r0:"
        data-state="closed"
>
  Open
</button>
`;
